actors/miner/tests/miner_actor_test_bitfield_queue.rs

```diff
@@ -0,0 +1,273 @@
+use fil_actor_miner::BitFieldQueue;
+use fil_actors_runtime::test_utils::MockRuntime;
+use fvm_ipld_amt::Amt;
+use fvm_ipld_bitfield::iter::Ranges;
+use fvm_ipld_bitfield::BitField;
+use fvm_ipld_blockstore::MemoryBlockstore;
+use fvm_shared::clock::{ChainEpoch, QuantSpec, NO_QUANTIZATION};
+use std::collections::HashMap;
+use std::ops::Range;
+
+mod util;
+use util::*;
+
+const TEST_AMT_BITWIDTH: u32 = 3;
+
+pub struct BQExpectation {
+    expected: HashMap<ChainEpoch, Vec<u64>>,
+}
+
+impl BQExpectation {
+    fn add(&mut self, epoch: ChainEpoch, values: Vec<u64>) -> &mut BQExpectation {
+        self.expected.entry(epoch).or_insert(values);
+        self
+    }
+
+    fn equals(&mut self, q: &mut BitFieldQueue<MemoryBlockstore>) {
+        assert_eq!(self.expected.len() as u64, q.amt.count());
+
+        q.amt
+            .for_each_mut(|epoch, bitfield| {
+                let values = &self.expected[&(epoch as i64)];
+                assert_bitfield_equals(bitfield, values);
```

Comment:
nit: by convention we put the expected value on the left and the actual value on the right

Reply:
That won't type-check in this case. When using `assert_bitfield_equals`, specs-actors also put the expected values on the right and the actual value on the left.